### PR TITLE
Mock exe info fix

### PR DESCRIPTION
**Pull request by marcvberg**
#### Description of changes

Adds some small fixes in mock startup to prevent breakage. Some null safety as well as an added call to constructExeInfo because it wasn't being populated properly during a partial mock push to cloud

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by kuhlmanp — DISMISSED** [review]

**Review by sdstolworthy — DISMISSED** [review]

**Comment by alharris-at** [comment]
Could you add an e2e test for this as well?

File: packages/amplify-util-mock/API.md [hunk 1]
```diff
@@ -3,11 +3,22 @@
...
+
 // @public (undocumented)
 export function executeAmplifyCommand(context: any): Promise<void>;
 
+// @public (undocumented)
+export class GraphQLClient {
+    constructor(url: string, headers: any);
+    // Warning: (ae-forgotten-export) The symbol "GraphQLResponse" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
```
Comment (sobolk) [inline packages/amplify-util-mock/API.md:15]: can you address this warning ?

File: packages/amplify-util-mock/src/index.ts [hunk 2]
```diff
@@ -1 +1,2 @@
 export * from './amplify-plugin-index';
+export { GraphQLClient } from './__e2e__';
```
Comment (sobolk) [inline packages/amplify-util-mock/src/index.ts:2]: should we export type from `__e2e__` package ?

**Review by sobolk — CHANGES_REQUESTED** [review]

**Comment by sobolk** [comment]
Btw. can you please avoid force pushes? It makes tracking incremental changes on PR harder.

**Comment by marcvberg** [comment]
@sobolk  Have resolved issues, sorry for the force push, I was doing a rebase. Is it preferred to just merge?

**Comment by sobolk** [comment]
> @sobolk Have resolved issues, sorry for the force push, I was doing a rebase. Is it preferred to just merge?

Yes, merges are preferred, we can then use GitHub UI to mark files as reviewed and review only recent diffs. Force pushes often invalidate state and reviewer has to start from scratch. So it's better to avoid it if possible.

File: packages/amplify-e2e-core/src/init/amplifyMock.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { ExecutionContext, getCLIPath, nspawn as spawn } from '..';
```
Comment (sobolk) [inline packages/amplify-e2e-core/src/init/amplifyMock.ts:1]: if `amplify mock` is top level command then should this file go to `packages/amplify-e2e-core/src/categories/mock.ts`

File: packages/amplify-util-mock/src/__e2e__/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export { GraphQLClient } from './utils';
```
Comment (sobolk) [inline packages/amplify-util-mock/src/__e2e__/index.ts:1]: is this still needed?

**Review by sobolk — COMMENTED** [review]
LGTM.

**Comment by sobolk** [comment]
Btw. you can use this command locally to validate that static analysis is happy
https://github.com/aws-amplify/amplify-cli/blob/7ed3c488b090943d4d74d29e8c4e3974a0dc531f/package.json#L7

File: packages/amplify-e2e-core/src/init/amplifyMock.ts (hunk 3 above)
Comment (marcvberg) [reply packages/amplify-e2e-core/src/init/amplifyMock.ts:1]: I'd say no, I don't consider mock to be a category, it's more of a subcommand applying to many categories. So when you run a generic mock, it's mocking all the other categories without specifically targeting one

File: packages/amplify-util-mock/src/__e2e__/index.ts (hunk 4 above)
Comment (marcvberg) [reply packages/amplify-util-mock/src/__e2e__/index.ts]: not needed, removed

**Review by sobolk — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Review by sdstolworthy — APPROVED** [review]

File: packages/amplify-e2e-tests/src/__tests__/mock/auth-transformer-mock.test.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    await addApiWithAllAuthModes(projRoot, { transformerVersion: 2 });
+    await updateApiSchema(projRoot, projName, 'simple_model_api_key.graphql');
+    executionContext = amplifyMock(projRoot);
+    graphqlClient = new GraphQLClient(endpoint, { 'x-api-key': apiKey });
+  });
+
+  afterEach(async () => {
+    await executionContext.sendCtrlC().runAsync();
+    await deleteProject(projRoot);
+    deleteProjectDir(projRoot);
```
Comment (manaswi223) [inline packages/amplify-e2e-tests/src/__tests__/mock/auth-transformer-mock.test.ts:31]: nit: extra line same as above. Do we need both of them (async & sync delete) ?
Or did you mean to do something like:
```

      const mockFilePath = path.join(projRoot, 'amplify', 'mock-data');
      if (existsSync(metaFilePath)) {
        await deleteProject(projRoot);
      }
      deleteProjectDir(projRoot);
     
```

**Pull request merged**